archivesunleashed/auk
# Add Scaling Design for SigmaJS -- partially address Issue 130

**The title of this pull-request should be a brief description of what the pull-request fixes/improves/changes. Ideally 50 characters or less.**

* * *

**GitHub issue(s)**:

#130

# What does this Pull Request do?

- Add a scaling feature to enable showing of more labels on graph.
- Make the transition from page to full screen more consistent (same graph shows in each place).
- Move default settings to instance rather than library.

# How should this be tested?

Spin up an instance with an account and use the "scale up" feature.  What should happen is that the next-to-largest nodes will get larger and therefore start to show labels.
Switching back-and-forth from fullscreen should show the exact same graph.

# Additional Notes:

The scale up checks whether each node is the same size as the largest node and if not, it squares the size value.  Sigma takes care of the scaling and displaying.

I tried "scale down" using sqrt, but the max-value check means scale down does not restore back from scale up.  I found that confusing, so just left "scale up" which can be restored to normal using refresh.

I'm sure there's a font-awesome icon for this feature instead of text, but I did not see it yet.  Icon options for bootstrap 4 are here:  https://fontawesome.bootstrapcheatsheets.com/  

# Interested parties

@ianmilligan1 @ruebot 

Thanks in advance for your help with the Archives Unleashed Toolkit!
